### ISSUE TITLE
Gaia plug-in interface tuning, release selection

### ISSUE THREAD

**Issue by mpyat2** [AAVSO/VStar#337]
Selection between Gaia DR2/DR3, parameter dialog persistence between the plug-in calls

**Comment by dbenn** [comment]
Is that meant to be "Gaia DR2/DR23" vs "Gaia DR2/DR2" @mpyat2 ?

I was wondering about that too and agree it's a good idea.

**Comment by dbenn** [comment]
I also think it would be useful to be able to load a Gaia file as well, using the standard obs source file chooser (+/- URL request button). What do you think?

**Comment by mpyat2** [comment]
Yes, I meant DR2/DR3. I'm working on it; see the branch "issue-337-gaia-plug-in-enchancement"; it is almost finished. I hope I'll make the pull request soon.
Concerning loading Gaia data from the file: as far as a separate standard ob-source plug-in is needed for this, it would be ok to share a common code part between the URL-based and file-based Gaia plug-ins. Currently, it is possible via the separate jar placed into vstar_plugin_libs. What do you think? [Such an approach could be also used for the ZTF plug-in]

**Comment by dbenn** [comment]
Sounds good Max

**Comment by mpyat2** [comment]
Well, things are going on. The common ancestor GaiaObSourceBase was moved to a shared jar placed in vstar_plugin_libs and the file observation source has been created. However, the PlugIn manager refused to work with this shared lib. So I changed it by taking the PluginLoader as a sample. It seems that some code can be shared between the PlugIn manager and the PluginLoader; some optimization can be made in the future. I'm continuing testing; the code has been pushed (see the branch issue-337-gaia-plug-in-enhancement) without the pull request though.
